src/borg/_hashindex.c

```diff
@@ -146,7 +146,10 @@ hashindex_lookup(HashIndex *index, const void *key, int *start_idx)
             }
             return idx;
         }
-        idx = (idx + 1) % index->num_buckets;
+        idx++;
+        if (idx >= index->num_buckets) {
+            idx = 0;
```

Comment:
hmm, this is a bit logically strange, without causing an actual malfunction.

i think it either should be `idx == index->num_buckets`, because only in that case `idx = 0` as action is correct AND we can be sure to hit that index as we are always incrementing by 1.

OR

keep `idx >= index->num_buckets`, but then it needs to be `idx -= index->num_buckets`.